src/nvim/api/buffer.c

```diff
@@ -738,6 +742,223 @@ ArrayOf(Integer, 2) nvim_buf_get_mark(Buffer buffer, String name, Error *err)
   return rv;
 }
 
+/// Returns extended mark info for a given mark identifier
+/// If no mark is found, returns an empty list.
+///
+/// If an invalid namespace or buffer is given, an error will be
+/// raised and 0 returned.
+///
+/// @param buffer The buffer handle
+/// @param namespace a identifier returned previously with nvim_create_namespace
+/// @param id any mark identifier that uniquely selects a mark (no positions)
+/// @param[out] err Details of an error that may have occurred
+/// @return [mark_id, row, col]
+ArrayOf(Object) nvim_buf_lookup_mark(Buffer buffer,
```

Comment:
"lookup" is new jargon without a clear reason. `get` or `get_by_foo` or `find` would be better.

`nvim_buf_get_mark` already exists, it is meaningless to differentiate API functions by synonyms ("get" vs "lookup"). If a new function has the same semantics as an existing one, one should just admit it and append a number (`nvim_buf_get_mark2`).

I would suggest `nvim_buf_get_ns_mark` or `nvim_buf_get_mark2`. "find" wouldn't be the right verb here because it's not a search, it's an index (AKA lookup, AKA `get`).

Reply:
Or `nvim_buf_get_mark_info`.

Reply:
I guess the problem is we have two things that are named marks. if we do xxx2, then there should be a relation between the two. In that case i would prefer to merge the old marks into the new marks api. Meaning a namespace and everything would be required. Another cleaner way woud be to introduce another name to differentiat extended marks and traditional marks.

Reply:
We probably want a proper noun to use consistently in both overview docs and API, to distinguish them from ordinary (named) marks. Either `extmark` or `nsmark` as was suggested, but if the later then we should call them "namespaced marks" and not "extended marks" in docs. `nvim_buf_get_mark2` is ambigous if its just the function that's different and not the marks, similarly with `mark_info` (which sounds more like returning a dict with info about marks in general, traditional and extended)

Reply:
Agreed. Let's stick with `extmark` for now since it's less chaotic for this PR; can change it to `nsmark` later if we want. 

So for this case:

    nvim_buf_get_extmark